pkg/controller/wildflyserver/wildflyserver_controller.go

```diff
@@ -130,12 +170,161 @@ func (r *ReconcileWildFlyServer) Reconcile(request reconcile.Request) (reconcile
+	scaleDownPodsState := make(map[string]string)                              // map referring to: pod name - pod state
+	scaleDownErrors := []error{}                                               // errors occured during processing the scaledown for the pods
+	for scaleDownIndex := 1; scaleDownIndex <= numberOfPodsToScaleDown; scaleDownIndex++ {
+		// Scaledown scenario, need to handle transction recovery
+		scaleDownPod := podList.Items[len(podList.Items)-scaleDownIndex]
+		scaleDownPodName := scaleDownPod.ObjectMeta.Name
+		scaleDownPodIP := scaleDownPod.Status.PodIP
+		if strings.Contains(scaleDownPodIP, ":") && !strings.HasPrefix(scaleDownPodIP, "[") {
+			scaleDownPodIP = "[" + scaleDownPodIP + "]" // for IPv6
+		}
+
+		// Setting-up the pod status - status is used to decide if the pod could be scaled (aka. removed from the statefulset)
+		wildflyServerSpecPodStatus := getWildflyServerPodStatusByName(wildflyServer, scaleDownPodName)
+		scaleDownPodsState[scaleDownPodName] = wildflyServerSpecPodStatus.State
+		if wildflyServerSpecPodStatus.State != wildflyv1alpha1.PodStateScalingDownClean {
+			// When state is not already processed with recovery and all data was cleaned then we mark pod to be dirty
+			scaleDownPodsState[scaleDownPodName] = wildflyv1alpha1.PodStateScalingDownDirty
+		}
+
+		if wildflyServerSpecPodStatus.State != wildflyv1alpha1.PodStateScalingDownClean {
+			// Removing the pod from the Service handling
+			scaleDownPod.ObjectMeta.Labels[labelWildflyOperatorInService] = "under-scale-down-processing"
+			if err = r.client.Update(context.TODO(), &scaleDownPod); err != nil {
+				reqLogger.Error(err, "Failed to update pod labels", "Pod name", scaleDownPod.ObjectMeta.Name, "Pod labels ", scaleDownPod.ObjectMeta.Labels)
+				scaleDownErrors = append(scaleDownErrors, err)
+				break
+			}
+
+			reqLogger.Info("Going for scaledown with pod", "pod name", scaleDownPod.ObjectMeta.Name, "IP address", scaleDownPodIP)
+
+			managementURL := fmt.Sprintf("http://%v:%v/management", scaleDownPodIP, httpManagementPort)
+			username := string(foundTxnRecoverySecret.Data["username"])
+			password := string(foundTxnRecoverySecret.Data["password"])
+			httpDigest := wildflyutil.HTTPDigest{URL: managementURL, Username: username, Password: password}
+
+			// Enabling recovery listener to speed up recovery processing
+			jsonResult, err := wildflyutil.ExecuteMgmtOp(&httpDigest, wildflyutil.MgmtOpTxnEnableRecoveryListener)
+			if err != nil {
+				reqLogger.Error(err, "Cannot enable transaction recovery listener", "Management URL", managementURL, "Pod name", scaleDownPod)
+				scaleDownErrors = append(scaleDownErrors, err)
+				break
+
+			}
+			if !wildflyutil.IsMgmtOutcomeSuccesful(jsonResult) {
+				reqLogger.Info("Failed to enable transaction recovery listener. Scaledown processing can take longer",
+					"Management command", wildflyutil.MgmtOpTxnEnableRecoveryListener, "JSON response", jsonResult)
```

Comment:
Is there a missing `break` if the op is not successful?

Reply:
or appending to `scaleDownErrors`?

Reply:
My idea was that if transaction recovery listener was not successfully set then the recovery will be still processing but slowly. Every 2 minutes the recovery will be kicked off. And thus there is chance that after 2 minutes the transactions will be processed.

I do code changes based on the reproaches from the discussion with QE guys and the logic needs to be adjusted and that version will put the error on this operation under `scaleDownErrors`.